client/engine/chainservice/mockchain.go

```diff
@@ -61,12 +61,15 @@ func (mc MockChain) handleTx(tx protocols.ChainTransaction) {
 		AdjudicationStatus: protocols.AdjudicationStatus{TurnNumRecord: 0},
 	}
 	for _, out := range mc.out {
-		sendEvent(out, event)
+		attemptSend(out, event)
 	}
 
 }
 
-// sendEvent sends event to the supplied chan
-func sendEvent(out chan Event, event Event) {
-	out <- event
+// attemptSend sends event to the supplied chan, and drops it if the chan is full
+func attemptSend(out chan Event, event Event) {
+	select {
+	case out <- event:
+	default:
```

Comment:
According to my understanding, we hit this `default` case here only when we have badly misconfigured some integration test. Do we want to:

- return an error
- panic

I'm loosely in favour of panic here, since it doesn't complicate the usage of `attemptSend`, and provides the fast feedback that things are wrong in the test setup.

Reply:
I anticipate we may do tests in the future where we manually "crash" a client, which would trigger the default case here. So I'm in two minds about whether we should panic or not. 

My feeling is that the code as-is is pretty close to a genuine "broadcast" or "fire and forget" pattern akin to a genuine blockchain, whereas putting in a panic would push us back towards a pattern where "messages being received properly is a _requirement_". To me the latter pattern is characteristic of communicating with `chans`, and not really something that fits the behaviour of what we are trying to mock, here.

Most likely this isn't of huge importance right now and we can amend it in future when we have more context?

Reply:
That sounds good. In practice a recovering client will have the ability to 'look up' past messages it has missed via a (mock) chain explorer.